diff --git a/src/adw-about-window.c b/src/adw-about-window.c
index e861dcd0..eda306bc 100644
--- a/src/adw-about-window.c
+++ b/src/adw-about-window.c
@@ -6,7 +6,10 @@
 
 #include "config.h"
 #include <glib/gi18n-lib.h>
+
+#ifndef G_OS_WIN32
 #include <appstream.h>
+#endif
 
 #include "adw-about-window.h"
 
@@ -388,12 +391,21 @@ activate_link_default_cb (AdwAboutWindow *self,
   return GDK_EVENT_STOP;
 }
 
+#ifdef G_OS_WIN32
+static gboolean
+get_release_for_version (void  *rel,
+                         const char *version)
+{
+  return false;
+}
+#else
 static gboolean
 get_release_for_version (AsRelease  *rel,
                          const char *version)
 {
   return !g_strcmp0 (as_release_get_version (rel), version);
 }
+#endif
 
 static void
 update_credits_legal_group (AdwAboutWindow *self)
@@ -1953,6 +1965,9 @@ GtkWidget *
 adw_about_window_new_from_appdata (const char *resource_path,
                                    const char *release_notes_version)
 {
+#ifdef G_OS_WIN32
+  return adw_about_window_new();
+#else
   AdwAboutWindow *self;
   GFile *appdata_file;
   char *appdata_uri;
@@ -1970,6 +1985,7 @@ adw_about_window_new_from_appdata (const char *resource_path,
   appdata_file = g_file_new_for_uri (appdata_uri);
 
   self = ADW_ABOUT_WINDOW (adw_about_window_new ());
+
   metadata = as_metadata_new ();
 
   if (!as_metadata_parse_file (metadata, appdata_file, AS_FORMAT_KIND_UNKNOWN, &error)) {
@@ -2081,6 +2097,7 @@ adw_about_window_new_from_appdata (const char *resource_path,
   g_free (appdata_uri);
 
   return GTK_WIDGET (self);
+#endif
 }
 
 /**
diff --git a/src/meson.build b/src/meson.build
index 1366bcc1..bc766501 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -286,23 +286,26 @@ gtk_dep = dependency('gtk4',
     'build-tests=false',
   ],
 )
-appstream_dep = dependency('appstream',
-  fallback : ['appstream', 'appstream_dep'],
-  default_options : [
-    'systemd=false', 'apidocs=false', 'install-docs=false',
-    'stemming=false', 'svg-support=false', 'gir=false',
-  ],
-)
 
 libadwaita_deps = [
   dependency('glib-2.0', version: glib_min_version),
   dependency('fribidi'),
   gio_dep,
   gtk_dep,
-  appstream_dep,
   cc.find_library('m', required: false),
 ]
 
+if host_machine.system() != 'windows'
+  appstream_dep = dependency('appstream',
+    fallback : ['appstream', 'appstream_dep'],
+    default_options : [
+      'systemd=false', 'apidocs=false', 'install-docs=false',
+      'stemming=false', 'svg-support=false', 'gir=false',
+    ],
+  )
+  libadwaita_deps += appstream_dep
+endif
+
 libadwaita_public_deps = [
   gio_dep,
   gtk_dep,
