diff --git a/demo/data/meson.build b/demo/data/meson.build
index 3638d4bc..1c14b1fc 100644
--- a/demo/data/meson.build
+++ b/demo/data/meson.build
@@ -28,25 +28,23 @@ appdata_config = configuration_data()
 appdata_config.set('BUILD_VERSION', meson.project_version())
 appdata_config.set('BUILD_DATE', today)
 
-if target_system != 'windows'
-  appstream_file = i18n.merge_file(
-    input: configure_file(
-      input: 'org.gnome.Adwaita1.Demo.metainfo.xml.in.in',
-      output: 'org.gnome.Adwaita1.Demo.metainfo.xml.in',
-      configuration: appdata_config
-    ),
-    output: 'org.gnome.Adwaita1.Demo.metainfo.xml',
-    po_dir: '../../po',
-    install: true,
-    install_dir: datadir / 'metainfo'
-  )
+appstream_file = i18n.merge_file(
+  input: configure_file(
+    input: 'org.gnome.Adwaita1.Demo.metainfo.xml.in.in',
+    output: 'org.gnome.Adwaita1.Demo.metainfo.xml.in',
+    configuration: appdata_config
+  ),
+  output: 'org.gnome.Adwaita1.Demo.metainfo.xml',
+  po_dir: '../../po',
+  install: true,
+  install_dir: datadir / 'metainfo'
+)
 
-  appstreamcli = find_program('appstreamcli', required: false)
-  if appstreamcli.found()
-    test('Validate appstream file', appstreamcli,
-      args: ['validate', '--no-net', '--explain', appstream_file]
-    )
-  endif
+appstreamcli = find_program('appstreamcli', required: false)
+if appstreamcli.found()
+  test('Validate appstream file', appstreamcli,
+    args: ['validate', '--no-net', '--explain', appstream_file]
+  )
 endif
 
 install_data(
diff --git a/src/adw-about-window.c b/src/adw-about-window.c
index 4bfbdc48..1446c3eb 100644
--- a/src/adw-about-window.c
+++ b/src/adw-about-window.c
@@ -6,7 +6,10 @@
 
 #include "config.h"
 #include <glib/gi18n-lib.h>
+
+#ifndef G_OS_WIN32
 #include <appstream.h>
+#endif
 
 #include "adw-about-window.h"
 
@@ -388,12 +391,21 @@ activate_link_default_cb (AdwAboutWindow *self,
   return GDK_EVENT_STOP;
 }
 
+#ifdef G_OS_WIN32
+static gboolean
+get_release_for_version (void  *rel,
+                         const char *version)
+{
+  return false;
+}
+#else
 static gboolean
 get_release_for_version (AsRelease  *rel,
                          const char *version)
 {
   return !g_strcmp0 (as_release_get_version (rel), version);
 }
+#endif
 
 static void
 update_credits_legal_group (AdwAboutWindow *self)
@@ -1963,6 +1975,9 @@ GtkWidget *
 adw_about_window_new_from_appdata (const char *resource_path,
                                    const char *release_notes_version)
 {
+#ifdef G_OS_WIN32
+  return adw_about_window_new();
+#else
   AdwAboutWindow *self;
   GFile *appdata_file;
   char *appdata_uri;
@@ -1980,6 +1995,7 @@ adw_about_window_new_from_appdata (const char *resource_path,
   appdata_file = g_file_new_for_uri (appdata_uri);
 
   self = ADW_ABOUT_WINDOW (adw_about_window_new ());
+
   metadata = as_metadata_new ();
 
   if (!as_metadata_parse_file (metadata, appdata_file, AS_FORMAT_KIND_UNKNOWN, &error)) {
@@ -2091,6 +2107,7 @@ adw_about_window_new_from_appdata (const char *resource_path,
   g_free (appdata_uri);
 
   return GTK_WIDGET (self);
+#endif
 }
 
 /**
diff --git a/src/meson.build b/src/meson.build
index 2a83a71c..29ba80d4 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -274,10 +274,16 @@ libadwaita_deps = [
   dependency('fribidi'),
   gio_dep,
   gtk_dep,
-  appstream_dep,
   cc.find_library('m', required: false),
 ]
 
+if host_machine.system() != 'windows'
+  appstream_dep = dependency('appstream',
+    fallback : ['appstream', 'appstream_dep']
+  )
+  libadwaita_deps += appstream_dep
+endif
+
 libadwaita_public_deps = [
   gio_dep,
   gtk_dep,
